mantid/wfm/wfm_stitching.md

```diff
@@ -0,0 +1,169 @@
+        if x >= frame_boundaries[n].lower and x < frame_boundaries[n].upper:
+            frame_number[b] = n
+            break
+
+# Perform correction to TOF
+for e in raw_events:
+    iframe = int((e - tmin) / bin_width)
+    e += frame_shifts[frame_number[iframe]]
+```
+This assumes that the number of events to be processed is much higher than `Nbins`. This can provide 5x to 10x speedups for the conversion to TOF. Some questions remain as to whether such an optimization is applicable to the case where different corrections are applied to different detector pixels if a very large number of pixels is present, as having 5000 different bins for 1M pixels does not fit in memory for common computers.
+
+**Links**
+
+- The software is available at: https://github.com/nvaytet/wfmess
+- Contact: Neil Vaytet (ESS/DMSC)
+
+## 4. In-reduction stitching vs post-processing stitching
+
+The early versions of the stitching procedure (including Method 1) depended on the Mantid software to perform the stitching, and consisted of several function calls that had to be made in a reduction workflow in order to stitch the data before further manipulation/reduction could be carried out. This was reported to be very time-consuming by several users, as they had to re-run a potentially lengthy stitching step, which was always performing the same calculations.
+
+In principle, stitching is only required to be run once, as very little tweaking of the stitching parameters should be necessary. In addition, some users were carrying out a study that was comparing WFM and single-pulse data using the same sample set-up, and were having to maintain two different (but very similar) reduction scripts, depending on whether they were reducing WFM or single-pulse data.
+
+We thus propose that WFM stitching could be implemented as a post-processing step in the data acquisition. The raw data could be read in, stitched using the parameters from the beamline (the Nexus geometry is contained inside the data file), and a new post-processed file would then be written to disk (and added to SciCat). All the parameters that were used for the stitching would be included in the file metadata. This would effectively alleviate all the issues listed above.
+
+The users would still have access to the raw un-stitched data, so that they are able to perform the stitching themselves, if they suspect something went wrong during that phase. But for 99% of cases, assuming auto-reduction is operating to satisfaction at an ESS facilty in full production mode, automatic post-processed stitching should work nicely.
+
+Going one step further, one could even imagine including the event-by-event stitching (although it makes less sense to call it 'stitching' here, 'conversion to time-of-flight' may be more appropriate) as part of the filewriter. It should be possible to write the raw and stitched data files simultaneously, thus cutting out the time needed to re-read the raw data to perform the conversion.
+
```

Comment:
WFM correction as part of the live stream is also very important. Using the TOF calculation approach based on beamline geometry, this should be possible. Do we bake that into the decoder?